lib/class-wp-block-type.php

```diff
@@ -22,6 +22,22 @@ class WP_Block_Type {
 	 */
 	public $name;
 
+	/**
+	 * Block type category name.
+	 *
+	 * @since 2.5.0
+	 * @var string
+	 */
+	public $category;
+
+	/**
+	 * Block type icon name.
+	 *
+	 * @since 2.5.0
+	 * @var string
+	 */
+	public $icon;
```

Comment:
This probably should stay on the client as it is UI related.

Reply:
IS this path to file or the SVG itself? If it's path to file, I think it's easier if it's done in PHP.

Reply:
I will have to double check, but I believe you can provide JSX there.

Reply:
It's just the name of the Dashicon to use.

Reply:
In core we use Dashicons, but it also works with SVG components, see https://github.com/WordPress/gutenberg/blob/master/blocks/block-icon/index.js.